### PR TITLE
Fantasy redgate map

### PR DESCRIPTION
**Pull request by SatinIsle**
Added a new fantasy redgate map that consists of a fantasy town and a large dungeon beneath it.

Added a selection of fantasy props.

Added a variety of "magic" resprites of existing items. These are intended to be tech disguised as magic and have descriptions to hint at it, along with some papers on the map.

Added an alchemy system that creates potions in an alembic out of an ingredient and a base. If the ingredient and base match, an interesting potion is created, if they don't, a poor quality potion is created.

Added a new reagent that polymorphs the drinker into a random creature.

Added a chest version of crates.

Added wooden wall lockers.

Added random spawners for fantasy items, potions, ingredients and bases for mapping.

Added hedge and wooden fences.

Added an old fashioned resprite of the oven.

Added a cooking pot resprite of the microwave.

Added wall torches.

Added barrel version of water tanks, beer tanks, wine tanks, blood tanks and a kettle resprite of a coffee dispenser.

Added a wooden tub bath.

Added 19 new vore mobs: 2 Catslugs, 6 succubi, 1 cryptdrake, 4 vampires 5 peasants and a bat.

### CODE REVIEW

**Comment by Heroman3003** [comment]
Okay, so, for organization purposes of things such as this-- It's best to create a code file within same folder as the map and put all the map-specific/exclusive items for that one map there, to avoid there being obscure things such as these additions strewn all over the code.

File: code/game/objects/items/devices/scanners/health.dm [hunk 1]
```diff
@@ -304,15 +304,14 @@
 			var/blood_volume = H.vessel.get_reagent_amount("blood")
 			var/blood_percent =  round((blood_volume / H.species.blood_volume)*100)
 			var/blood_type = H.dna.b_type
-			var/blood_reagent = H.species.blood_reagents
```
Comment (Heroman3003) [inline code/game/objects/items/devices/scanners/health.dm:307]: Looks like you are accidentally removing a feature here in this file

Comment (SatinIsle) [reply code/game/objects/items/devices/scanners/health.dm:307]: Yep, just spotted that one, can readd that real quick

**Comment by SatinIsle** [comment]
> Okay, so, for organization purposes of things such as this-- It's best to create a code file within same folder as the map and put all the map-specific/exclusive items for that one map there, to avoid there being obscure things such as these additions strewn all over the code.

I've combined most of the ones that otherwise feel out of place in their respective files now, which is pretty much all the resprited ones.

**Pull request merged**